server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keyword.java

```diff
@@ -0,0 +1,126 @@
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.model;
+
+import java.util.Optional;
+import javax.mail.Flags;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.base.CharMatcher;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableBiMap;
+import com.google.common.collect.ImmutableList;
+
+public class Keyword {
+    private final static int FLAG_NAME_MIN_LENTH = 1;
+    private final static int FLAG_NAME_MAX_LENTH = 255;
+    private final static CharMatcher FLAG_NAME_PATTERN = CharMatcher.JAVA_LETTER_OR_DIGIT.or(CharMatcher.is('$'));
+
+    public final static Keyword DRAFT = new Keyword("$Draft");
+    public final static Keyword SEEN = new Keyword("$Seen");
+    public final static Keyword FLAGGED = new Keyword("$Flagged");
+    public final static Keyword ANSWERED = new Keyword("$Answered");
+    public final static Keyword DELETED = new Keyword("$Deleted");
+    public final static Keyword RECENT = new Keyword("$Recent");
+    public final static Boolean FLAG_VALUE = true;
+
+    private final static ImmutableList<Keyword> NON_EXPOSED_IMAP_KEYWORDS = ImmutableList.of(Keyword.RECENT, Keyword.DELETED);
+    private final static ImmutableBiMap<Flags.Flag, Keyword> IMAP_SYSTEM_FLAGS = ImmutableBiMap.<Flags.Flag, Keyword>builder()
+        .put(Flags.Flag.DRAFT, DRAFT)
+        .put(Flags.Flag.SEEN, SEEN)
+        .put(Flags.Flag.FLAGGED, FLAGGED)
+        .put(Flags.Flag.ANSWERED, ANSWERED)
+        .put(Flags.Flag.RECENT, RECENT)
+        .put(Flags.Flag.DELETED, DELETED)
+        .build();
+
+    private final String flagName;
+
+    public static Keyword fromFlag(Flags.Flag flag) {
+        return IMAP_SYSTEM_FLAGS.get(flag);
+    }
+
+    public Keyword(String flagName) {
+        Preconditions.checkArgument(isValid(flagName),
+                "Flagname must not be null or empty, must have length form 1-255, must not contain charater with hex from '\u0000' to '\u00019' or {'(' ')' '{' ']' '%' '*' '\"' '\\'} ");
+        this.flagName = flagName;
+    }
+
+    private boolean isValid(String flagName) {
+        if (StringUtils.isBlank(flagName)) {
+            return false;
+        }
+        if (flagName.length() < FLAG_NAME_MIN_LENTH || flagName.length() > FLAG_NAME_MAX_LENTH) {
+            return false;
+        }
+        if (!FLAG_NAME_PATTERN.matchesAllOf(flagName)) {
+            return false;
+        }
+        return true;
+    }
+
+    public String getFlagName() {
+        return flagName;
+    }
+
+    public boolean isNonExposedImapKeyword() {
+        return !NON_EXPOSED_IMAP_KEYWORDS.contains(this);
```

Comment:
Changed the method name to "isNotNonExposedImapKeyword" to meet its purpose
And yes, those are tested on KeywordTest, KeywordsTest and GetMessageListTest.getMessageListHasKeywordShouldIgnoreDeleted